src/main/java/com/epam/grid/engine/provider/job/slurm/SlurmJobProvider.java

```diff
@@ -144,9 +157,51 @@ public Job runJob(final JobOptions options) {
         return buildNewJob(getResultOfExecutedCommand(simpleCmdExecutor, makeSbatchCommand(options)));
     }
 
+    /**
+     * Deletes the job being performed according to the specified parameters.
+     *
+     * @param deleteJobFilter Search parameters for the job being deleted.
+     * @return Information about the deleted job.
+     */
     @Override
     public DeletedJobInfo deleteJob(final DeleteJobFilter deleteJobFilter) {
-        throw new UnsupportedOperationException("Job deletion operation haven't implemented yet");
+        if (!StringUtils.hasText(deleteJobFilter.getUser()) && deleteJobFilter.getId() == null) {
+            throw new GridEngineException(HttpStatus.BAD_REQUEST,
+                    String.format("Incorrect filling in %s. Either `id` or `user` should be specified for job removal!",
+                            deleteJobFilter));
+        }
+        if (deleteJobFilter.getId() != null && deleteJobFilter.getId() <= 0) {
+            throw new GridEngineException(HttpStatus.BAD_REQUEST,
+                    String.format("Id specified in %s for job removal is invalid!", deleteJobFilter));
+        }
+
+        final String deletingUser;
```

Comment:
I think that `jobOwner` for the name would be better

Reply:
yes, you're right